File: app/src/main/java/org/deiverbum/app/BaseApplication.java
package org.deiverbum.app;


import android.app.Application;

import androidx.hilt.work.HiltWorkerFactory;
import androidx.work.Configuration;

import com.google.firebase.crashlytics.FirebaseCrashlytics;

import javax.inject.Inject;

import dagger.hilt.android.HiltAndroidApp;

/**
 * Created by A. Cedano on 11,November,2021
 */
@HiltAndroidApp
public class BaseApplication extends Application implements Configuration.Provider {

    @Inject
    HiltWorkerFactory workerFactory;

    @Override
    public Configuration getWorkManagerConfiguration() {
        return new Configuration.Builder()
                .setWorkerFactory(workerFactory)
                .build();
    }


}

File: app/src/main/java/org/deiverbum/app/BaseApplication.kt
package org.deiverbum.app

import android.app.Application
import dagger.hilt.android.HiltAndroidApp
import timber.log.Timber

/**
 * Created by A. Cedano on 11,November,2021
 */
@HiltAndroidApp
class BaseApplication : Application() {
    //@Inject
    //lateinit var imageLoader: dagger.Lazy<ImageLoader>

    /*
        @Inject
        lateinit var profileVerifierLogger: ProfileVerifierLogger
    */
    override fun onCreate() {
        super.onCreate()
        if (BuildConfig.DEBUG) {
            Timber.plant(Timber.DebugTree())
        }
        // Initialize Sync; the system responsible for keeping data in the app up to date.
        //Sync.initialize(context = this)
        //profileVerifierLogger()
    }

    //override fun newImageLoader(): ImageLoader = imageLoader.get()
}
